orchagent/routeorch.cpp

```diff
@@ -688,6 +704,32 @@ void RouteOrch::doTask(Consumer& consumer)
                     {
                         nhg = NextHopGroupKey();
                     }
+                    else if (srv6_nh == true)
+                    {
+                        string ip;
+                        if (ipv.empty())
+                        {
+                            ip = "0.0.0.0";
+                        }
+                        else
+                        {
+                            SWSS_LOG_ERROR("For SRV6 nexthop ipv should be empty");
+                            continue;
+                        }
+                        nhg_str = ip + NH_DELIMITER + srv6_segv[0] + NH_DELIMITER + srv6_src[0];
+
+                        for (uint32_t i = 1; i < srv6_segv.size(); i++)
+                        {
+                            if (ipv.empty()) {
+                                ip = "0.0.0.0";
+                                nhg_str += NHG_DELIMITER + ip;
+                            }
+                            nhg_str += NH_DELIMITER + srv6_segv[i];
+                            nhg_str += NH_DELIMITER + srv6_src[i];
+                        }
+                        nhg = NextHopGroupKey(nhg_str, overlay_nh, srv6_nh);
+                        SWSS_LOG_NOTICE("SRV6 route with nhg %s", nhg.to_string().c_str());
+                    }
```

Comment:
I still don't like this block. 
Can be simplified as :
if ipv not empty then send error log and continue. 
Else 
for loop
nhg_str += "0.0.0.0" + .......
 
This should be much easier to read.

Reply:
Thanks..Done

Reply:
why should nhg_str contain "0.0.0.0"?